daemon.go

```diff
@@ -175,17 +175,17 @@ func (d *daemon) runCheck(query url.Values) (*output, error) {
 		out.DataAvailableOverBitswap.Error = "could not connect to peer"
 	} else {
 		// If so is the data available over Bitswap?
-		out.DataAvailableOverBitswap = checkBitswapCID(ctx, c, ma)
+		out.DataAvailableOverBitswap = checkBitswapCID(ctx, testHost, c, ma)
```

Comment:
Why would we do this? Seems the same as the discussion in this thread https://github.com/ipfs-shipyard/ipfs-check/pull/35#discussion_r1692875885

By passing the testHost after having dialed with all the addresses in `ai` means we still haven't tested whether the given multiaddress is actually dialable or not.

Reply:
We only append the resolved maddrs from the DHT to `ai` **if no `maddr` is passed in**:

https://github.com/ipfs-shipyard/ipfs-check/blob/65ee5cbb63e82faf6645961725421be3312336ca/daemon.go#L142-L155

This means that that if an `maddr` was passed and `testHost` has successfully connected, it's necessarily using the passed maddr.

Reply:
Thanks, wasn't paying close enough attention that should work 🙏.